### PR TITLE
feat(web-app): migrate status indicator table

### LINKED ISSUE #570: Add the StatusIndicatorTable MDX component from Carbon website

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #570 

I might've one more grid than needed here, wrapped the whole thing in a condensed grid to get grid alignment, suggestions welcome!

#### Testing / reviewing

http://localhost:3000/mdx/status-indicator-table
https://v10.carbondesignsystem.com/patterns/status-indicator-pattern/#high-attention

### CODE REVIEW

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Only some small fixes, this looks great otherwise. Wrapping the grid like that looks good to me to get the correct "hang". There is definitely some crazy stuff going on in this component to get it to sit on the grid, but seems like its working.

File: services/web-app/components/status-indicator-table/status-indicator.module.scss [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+.table .cell,
+.table .headerCell {
+  padding: spacing.$spacing-05;
+  color: theme.$text-primary;
+  vertical-align: top;
+}
+
+.table :is(.cell) {
+  border-collapse: collapse;
+  color: var(--cds-text-01);
```
[the remark this answers is not in the document]
Comment (francinelucca) [reply services/web-app/components/status-indicator-table/status-indicator.module.scss]: no idea, changed for token 👍

**Review by alisonjoseph — DISMISSED** [review]

File: services/web-app/components/status-indicator-table/status-indicator-row.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import {
+  cell,
+  descriptionCell,
+  glyph,
+  icon,
+  iconGroup,
+  iconWrapper,
+  isDark,
+  statusIndicatorRow
+} from './status-indicator.module.scss'
```
Comment (alisonjoseph) [inline services/web-app/components/status-indicator-table/status-indicator-row.js:21]:
```suggestion
import styles from './status-indicator.module.scss'
```

Comment (alisonjoseph) [reply services/web-app/components/status-indicator-table/status-indicator-row.js:21]: then update the file below to be styles. for all the class names

Comment (francinelucca) [reply services/web-app/components/status-indicator-table/status-indicator-row.js]: Done!

File: services/web-app/components/status-indicator-table/status-indicator-table.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import React from 'react'
+
+import yaml from '@/data/status-indicators/status-indicators.yaml'
+
+import {
+  container,
+  headerCell,
+  statusIndicatorTableWrapper,
+  table
+} from './status-indicator.module.scss'
```
[the remark this answers is not in the document]
Comment (francinelucca) [reply services/web-app/components/status-indicator-table/status-indicator-table.js]: Done!

**Review by alisonjoseph — DISMISSED** [review]
🚀

File: services/web-app/components/status-indicator-table/status-indicator.module.scss [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  border-bottom: 1px solid theme.$border-subtle;
+}
+
+.status-indicator-row:last-child {
+  border-bottom: none;
+}
+
+.status-indicator-table-wrapper {
+  margin-top: spacing.$spacing-06;
+  //Needed for Safari
```
Comment (andreancardona) [inline services/web-app/components/status-indicator-table/status-indicator.module.scss:105]:
```suggestion
  // Needed for Safari
```

**Review by andreancardona — APPROVED** [review]
tested and looks great!

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**